File: cpp_solution/language_learning/1161.cpp
#include<stdio.h>

char input_line[1000001];
int next[1000001];


void getNext(char input_string[], int result[], int str_len) {
    result[0] = -1;
    int j = 0;
    int k = -1;
    while (j < str_len) {
        if (k == -1){
            k++;
            j++;
            result[j] = k;
        } else if (input_string[j] == input_string[k]) {
            k++;
            j++;
            result[j] = k;
        } else {
            k = result[k];
        }
    }
}



int main() {
  while (scanf("%s", input_line) != EOF) {
	  if (input_line[0] == '.')  break;
	  else {
		  int input_len = 0;
		  for (input_len = 0; input_line[input_len]; input_len++)
			  ;
		  getNext(input_line, next, input_len);
		  int result_tmp = input_len/(input_len - next[input_len]);
		  if (  input_len%(input_len - next[input_len]) == 0) {
			  printf("%d\n", result_tmp);

		  } else {
			  printf("1\n");
		  }
	  }
  }
}

File: python_solution/language_learning/1161.py
def getNext(input_string, str_len):
    result = []
    result.append(-1)
    j = 0
    k = -1
    while j < str_len :
        if k == -1:
            k = k+1
            j = j+1
            result.append(k)
        elif input_string[j] == input_string[k]:
            k = k+1
            j = j+1
            result.append(k)
        else:
            k = result[k]
    return result
    




i=0
result = []
while 1:
    n = int(input())
    if n ==0:
        break
    input_line = input()
    input_len = len(input_line)

    next = getNext(input_line, input_len)
    
    if i != 0:
        print('')
    print("Test case #%d" % (i+1))

    j = 2
    while j <= n:
        repeat_len = j -next[j]
        if (j %repeat_len == 0) & ((int(j/repeat_len))!=1):
            print (j, int(j/repeat_len))
        j = j+1
    i = i+1